storage/remote/generic/generic.go

```diff
@@ -62,16 +69,15 @@ func (c *Client) Store(samples model.Samples) error {
 		}
 		req.Timeseries = append(req.Timeseries, ts)
 	}
-	client := NewGenericWriteClient(c.conn)
 	ctxt, _ := context.WithTimeout(context.Background(), c.timeout)
```

Comment:
Yeah, there's a lot of interesting questions about contexts - where to begin and terminate them, and whether it's worth threading them through everything or just parts of the call chain... also, on what level to apply timeouts: e.g. here in the caller of `Store()` or in `Store()` itself.

For now I think I'll create a TODO context directly in the generic `Write()` implemenation and set the timeout on that, since it's a timeout that's specifically configured for that implementation (maybe makes sense to change it once we really remove the others).